### PR TITLE
Remove API deprecated in 3.1 (part 2)

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Follow up and goes on top of to #15878. The are so many changes that I didn't want to put everything in one PR. Even in two parts, each part is quite long.

You may wait until #15878 is merged or just review the second commit in this PR.

### CODE REVIEW

**Comment by timhoffm** [comment]
For simplicity, will only fix the merge conflict after #15878 is in.

File: doc/api/next_api_changes/removals.rst [hunk 1]
```diff
@@ -1,2 +1,145 @@
...
+
+- ``axisartist.axis_artist.UnimplementedException`` (no replacement)
+- ``axisartist.axislines.SimpleChainedObjects`` (no replacement)
+- ``axisartist.axislines.Axes.AxisDict`` (no replacement)
+
+- ``testing.decorators.switch_backend()`` (use ``pytest.mark.backend`` instead)
+
+Others
+~~~~~~
+- The math text command ``\tackrel`` is removed. Use ``\genfrac`` instead.
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: stackrel

File: doc/api/next_api_changes/removals.rst [hunk 2]
```diff
@@ -1,2 +1,145 @@
...
+
+- ``ticker.OldScalarFormatter.pprint_val()`` (no replacement)
+- ``ticker.ScalarFormatter.pprint_val()`` (no replacement)
+- ``ticker.LogFormatter.pprint_val()`` (no replacement)
+- ``ticker.decade_down()`` (no replacement)
+- ``ticker.decade_up()`` (no replacement)
+
+- ``scale.LogTransformBase()`` (use ``scale.LogTransform()`` instead)
+- ``scale.InvertedLogTransformBase()`` (use ``scale.InvertedLogTransform()`` instead)
+- ``scale.Log10TransformBase()`` (use ``scale.Log10Transform()`` instead)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: wrong name

File: doc/api/next_api_changes/removals.rst [hunk 3]
```diff
@@ -1,2 +1,145 @@
...
+- ``scale.InvertedLog10TransformBase()`` (use ``scale.InvertedLog10Transform()`` instead)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: wrong name

File: doc/api/next_api_changes/removals.rst [hunk 4]
```diff
@@ -1,2 +1,145 @@
...
+- ``textpath.TextPath.is_math_text()`` (no replacement)
+- ``textpath.TextPath.text_get_vertices_codes()``
+  (use ``textpath.text_to_path.get_text_path()`` instead)
+
+- ``widgets.SpanSelector.buttonDown`` property (no replacement)
+
+- ``mplot3d.proj3d.line2d()`` (no replacement)
+- ``mplot3d.proj3d.line2d_dist()`` (no replacement)
+- ``mplot3d.proj3d.line2d_seg_dist()`` (no replacement)
+- ``mplot3d.proj3d.mod()`` (no replacement)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: this one had a replacement -- np.linalg.norm

**Comment by anntzer** [comment]
breakage is real

**Comment by timhoffm** [comment]
The failure is due to incomplete removal of `TextWithDash`. Since removing that is a bit more involved, I've pulled this out to a separate PR #15950. Will come back here once that is merged.

File: doc/api/next_api_changes/removals.rst [hunk 5]
```diff
@@ -71,7 +73,80 @@ Classes and methods
...
+
+- ``mlab.demean()`` (use ``mlab.detrend_mean()`` instead)
+- ``projections.process_projection_requirements()`` (no replacement)
+- ``path.get_paths_extents()``
+(use ``path.get_path_collection_extents()`` instead)
+
+- ``docstring.Appender`` (no replacement)
+- ``docstring.dedent()`` (use ``inspect.getdoc()`` instead)
+- ``docstring.copy_dedent()``
+  (use ``docstring.copy()`` and ``cbook.getdoc()`` instead)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: inspec.getdoc?

File: doc/api/next_api_changes/removals.rst [hunk 6]
```diff
@@ -71,7 +73,80 @@ Classes and methods
...
+
+- ``ticker.OldScalarFormatter.pprint_val()`` (no replacement)
+- ``ticker.ScalarFormatter.pprint_val()`` (no replacement)
+- ``ticker.LogFormatter.pprint_val()`` (no replacement)
+- ``ticker.decade_down()`` (no replacement)
+- ``ticker.decade_up()`` (no replacement)
+
+- ``scale.LogTransformBase()`` (use ``scale.LogTransform()`` instead)
+- ``scale.InvertedLogTransformBase()`` (use ``scale.InvertedLogTransform()`` instead)
+- ``scale.Log10TransformBase()`` (use ``scale.LogTransform()`` instead)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: names are off

File: lib/matplotlib/tests/test_scale.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 from matplotlib.cbook import MatplotlibDeprecationWarning
 import matplotlib.pyplot as plt
-from matplotlib.scale import (Log10Transform, InvertedLog10Transform,
-                              SymmetricalLogTransform)
+from matplotlib.scale import (LogTransform, InvertedLogTransform,
```
Comment (anntzer) [inline lib/matplotlib/tests/test_scale.py]: move all imports to second line?

File: doc/api/next_api_changes/removals.rst [hunk 8]
```diff
@@ -71,7 +73,80 @@ Classes and methods
...
+- ``scale.InvertedLog10TransformBase()`` (use ``scale.InvertedLogTransform()`` instead)
+- ``scale.Log2Transform()`` (use ``scale.LogTransform()`` instead)
+- ``scale.InvertedLog2Transform()`` (use ``scale.InvertedLogTransform()`` instead)
+- ``scale.NaturalLogTransform()`` (use ``scale.LogTransform()`` instead)
+- ``scale.InvertedNaturalLogTransform()`` (use ``scale.InvertedLogTransform()`` instead)
+
+- ``spines.Spine.is_frame_like()`` (no replacement)
+- ``text.is_math_text()`` (no replacement)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: cbook.is_math_text (below as well)

File: doc/api/next_api_changes/removals.rst [hunk 9]
```diff
@@ -71,7 +73,80 @@ Classes and methods
...
+- ``mplot3d.art3d.paths_to_3d_segments_with_codes()`` (no replacement)
+- ``mplot3d.art3d.get_patch_verts()`` (no replacement)
+- ``mplot3d.art3d.get_colors()`` (no replacement)
+- ``mplot3d.art3d.zalpha()`` (no replacement)
+
+- ``mplot3d.axis3d.get_flip_min_max()`` (no replacement)
+- ``mplot3d.axis3d.Axis.get_tick_positions()`` (no replacement)
+
+- ``axisartist.axis_artist.UnimplementedException`` (no replacement)
+- ``axisartist.axislines.SimpleChainedObjects`` (no replacement)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: axis_grid1.mpl_axes.SimpleChainedObjects (sorry)

File: doc/api/next_api_changes/removals.rst [hunk 10]
```diff
@@ -71,7 +73,80 @@ Classes and methods
...
+- ``axisartist.axislines.Axes.AxisDict`` (no replacement)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: axis_grid1.mpl_axes.Axes.AxisDict

File: doc/api/next_api_changes/removals.rst (hunk 9 above)
Comment (timhoffm) [reply doc/api/next_api_changes/removals.rst]: That's not officially deprecated.

File: doc/api/next_api_changes/removals.rst (hunk 10 above)
Comment (timhoffm) [reply doc/api/next_api_changes/removals.rst]: Also not deprecated.

File: doc/api/next_api_changes/removals.rst (hunk 9 above)
Comment (anntzer) [reply doc/api/next_api_changes/removals.rst]: I meant these are replacements (below too).

**Comment by anntzer** [comment]
Failures are real (bunch of mlab tests have to be deleted, in particular).

**Comment by timhoffm** [comment]
Still some math-text related image comparison failures.

**Comment by anntzer** [comment]
IIRC I had looked into this some time ago and the deprecation of stackrel may be the problematic one? in which case we can defer the removal of that part for now...

**Comment by timhoffm** [comment]
I don't think it's `\stackrel`. This PR removes the command, but there were no usages of that anymore.

**Comment by timhoffm** [comment]
Going to slice this down into smaller chunks to find out what is acutally causing the failure.

**Comment by anntzer** [comment]
I think #16204/#16206 takes care of mathtext.

**Comment by timhoffm** [comment]
Replaced by the sequence of above PRs.

**Pull request closed without merging**